indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```
@@ -197,7 +206,19 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
           activeTasks.size(),
           storedLocks.size() - taskLockCount
       );
+
+      // unlock all TaskLockPosse created, and also remove lock entries from storage
+      for (Task task : failedToReacquireLockTasks) {
+        unlockAll(task);
+        for (TaskLock lock : taskStorage.getLocks(task.getId())) {
+          taskStorage.removeLock(task.getId(), lock);
+        }
```

should this all be in a block where the exceptions can be ignored?